src/compiler/scala/tools/nsc/typechecker/Infer.scala

```diff
@@ -1396,13 +1407,25 @@ trait Infer extends Checkable {
           case tp               => tp
         }
 
+        // if at least one retry is required, and the overload includes a Java-defined alternative
+        // proceed without tupling for applicability (if flag is enabled)
+        private[this] var retries = 0
+        private def atRetryDepth[A](cond: Boolean)(body: => A): A = {
+          val res =
+            if (cond && retries > 0) withoutTuplingForApplicability(body)
+            else body
+          retries += 1
+          res
+        }
```

Comment:
I can't get ahold of the guy. He might be either out of the country or deep in-country. It's too bad he didn't leave a few more tests, as clues to what he might have been thinking.

It looks like you get one chance for your Java method to be chosen: implicits are off and `pt` is what it is. Thereafter, you don't want expected type messing with overload resolution involving a Java-defined method, perhaps the way you don't want an ump or ref's call deciding a sports match in the final moments. I'll see if there's a test case.